### PR TITLE
Add outputs to the action

### PR DESCRIPTION
**Pull request by 1fexd**
This adds outputs to the action, allowing other steps in the workflow to read information about the if there were actually any new commits, allows them to get the new commits, and if yes-create-issues is set to to, also a the list of issue created.

### CODE REVIEW

File: src/setActionOutput.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export function setActionOutput(createdIssues: CreatedIssue[], newCommits: Commit[]) {
+    const result: ActionOutputResult = { newCommitsInSyncRepo: newCommits.length > 0, newCommits: JSON.stringify(newCommits) }
+
+    core.setOutput("newCommitsInSyncRepo", result.newCommitsInSyncRepo)
+    core.setOutput("newCommits", result.newCommits)
+
+    if (createdIssues.length > 0) {
+        result.createdIssues = JSON.stringify(createdIssues)
```
Comment (iliabylich) [inline src/setActionOutput.ts:17]: Shouldn't these two lines be executed unconditionally?

**Review by iliabylich — COMMENTED** [review]
LGTM overall, just one question

File: src/setActionOutput.ts (hunk 1 above)
Comment (iliabylich) [reply src/setActionOutput.ts:17]: Ah, sorry, I think I'm reading it wrong. You always expose new commits, the only conditional part here is the array of created issues which can be empty if the action is configured to run in a dummy mode. 👍

**Comment by iliabylich** [comment]
Thank you

**Comment by 1fexd** [comment]
Well thank you, this was exactly what I was looking for, and your code is very good IMO (I don't personally really like JS/TS, and don't have a lot of experience with the ecosystem, but I think am going to use your action as template if I need to create a new custom one in the future)! 
Just one more thing: Could you do a release please, so it is available on Github Marketplace? Thanks!

**Comment by iliabylich** [comment]
Are you asking to cut a release or to make it available on the markerplace? The former is [already done, automatically](https://github.com/poll-github-repo/action/blob/master/.github/workflows/release.yml), pushing to the same v1 tag is not the best approach but so far I don't see how this action can get any breaking changes. The latter has been [done a long time ago](https://github.com/marketplace/actions/track-and-record-changes-in-other-repos)

**Comment by 1fexd** [comment]
Very strange that the repo has to include the dist output for Github actions to work.

**Comment by iliabylich** [comment]
Is it? Node is unable to run TS files (unless you are a "cool kid" on Deno or Bun.js 😄), so pre-compilation step is required. Literally every GH action that is not written in plain JS and/or uses new JS syntax has a `dist` directory or some variation of it in the built tag/branch that other people refer to in their workflows. The main/master branch of course doesn't have to have it.

**Comment by 1fexd** [comment]
Yeah, that's strange, like optimally on when you create a tag, and then a release from the tag it would build it action, then use just the resulting index.js as the "binary" for the action

**Comment by iliabylich** [comment]
Releases (i.e. GH releases) don't really matter in this case, when you add `- uses: org/repo@tag` you refer to a tag or a branch. A release with pre-built "archive" of everything (as an artifact) would be easier to work with, I agree, but it is what it is.

**Pull request merged**